Source/WarpX.cpp

```diff
@@ -1311,13 +1317,6 @@ WarpX::ReadParameters ()
             );
         }
 
-        if (J_in_time == JInTime::Constant)
-        {
-            WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
-                rho_in_time == RhoInTime::Linear,
-                "psatd.J_in_time=constant supports only psatd.rho_in_time=linear");
-        }
```

Comment:
Should we keep this in the case where we use the second-order system?

Reply:
I think #3506 will introduce the support for the options with both sources constant in time.